drivers/wifi/winc1500/wifi_winc1500.c

```diff
@@ -298,19 +298,21 @@ static int winc1500_get(sa_family_t family,
 			struct net_context **context)
 {
 	struct socket_data *sd;
+	SOCKET sock;
 
 	if (family != AF_INET) {
 		LOG_ERR("Only AF_INET is supported!");
 		return -1;
 	}
 
-	(*context)->offload_context = (void *)(sint32)socket(family, type, 0);
-	if ((*context)->offload_context < 0) {
+	sock = socket(family, type, 0);
+	if (sock < 0) {
 		LOG_ERR("socket error!");
 		return -1;
 	}
 
-	sd = &w1500_data.socket_data[(int)(*context)->offload_context];
+	(*context)->offload_context = (void *)(sint32)sock;
```

Comment:
It's not immediately (or even after search) obvious to me what type `SOCKET` is, but the above should be either:

    (void*)(intptr_t)sock

or

    (void*)(uintptr_t)sock

depending on whether `SOCKET` is a signed type.

Reply:
It is a `sint8`, which is a `signed char`. It is defined by the driver included in the atmel hal module, not zephyr. That said, I'll swap it out for `intptr_t`.

Reply:
Thanks, that explains why I couldn't find it.

Reply:
@legoabram could you push the update you described, please, so I can remove my NAK and we can merge this?  Thanks!

> It is a `sint8`, which is a `signed char`. It is defined by the driver included in the atmel hal module, not zephyr. That said, I'll swap it out for `intptr_t`.

Reply:
@pabigot Sorry about that! Got a bit distracted by work. Just rebased and pushed again, hopefully it passes.